datastore/save.go

```diff
@@ -67,8 +67,14 @@ func reflectFieldSave(props *[]Property, p Property, name string, opts saveOpts,
 		case reflect.Interface:
 			// Extract the interface's underlying value and then retry the save.
 			// See issue https://github.com/googleapis/google-cloud-go/issues/1474.
-			v = v.Elem()
-			return reflectFieldSave(props, p, name, opts, v)
+			if v.IsNil() {
+				// Nil interface becomes a nil property value (unless omitempty,
+				// handled by the embracing saveStructProperty call).
```

Comment:
How about this:
```
// Nil interface becomes a nil property value (unless omitEmpty is true,
// which is handled by the caller).
```

Please change L88 to say the same thing for a nil pointer as well. Thanks!

Reply:
No problem, pushed